homeassistant/components/binary_sensor/ihc.py

```diff
@@ -49,9 +55,13 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         binary_sensors = config[CONF_BINARY_SENSORS]
         for sensor_cfg in binary_sensors:
             ihc_id = sensor_cfg[CONF_ID]
+            controller_id = sensor_cfg[CONF_CONTROLLER]
+            ihc_controller = hass.data[controller_id][IHC_CONTROLLER]
```

Comment:
This is very brittle. It's easy for the user to make a mistake when naming the controllers which would break this.

Maybe we should make a breaking change by moving the platform configs to the component under the controller section? That would make it easier for the users and we could better guard against breakage.

Reply:
Yes! What do you think about that?

Reply:
Well.. mostly good. It will be more "clean" and remove the need of naming the controllers and keeping the configuration together. 
My biggest (only) worry is if we break any conventions on configuring binary_sensor, light, switch etc?

Reply:
(and it being a breaking change ;-)

Reply:
We now recommend to keep all configuration in the component section, so the change would be a step in the right direction in my opinion.

This will be a bigger change than initially planned. Is that ok with you time wise?

Reply:
Great to hear that it is the direction things are moving ;-)
I'm okay time wise with changing it. Thanks for asking!

Reply:
Is there any projects already following the new conventions?
It would be great with a reference in order to slice the schemas correctly

Reply:
Yes, look at octoprint:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/octoprint.py
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/binary_sensor/octoprint.py

Reply:
Give it a look. I have moved the device configuration to the component section and extended options for manual configuration. I have not yet updated the documentation but will do as soon as you have confirmed. The controllers and devices can now be configured like below:
```yaml
ihc:
  - url: 'http://192.168.1.3'
    username: YOUR_USERNAME
    password: YOUR_PASSWORD
    info: true 
    binary_sensor:
      38490: 
        name: switch_front_door
        inverting: false
        note: Magnet switch in door
        type: door
     64819: 
     ...
  - url: 'http://192.168.1.4'
     ...
```